CoEpi/native/LibResult.swift

```diff
@@ -0,0 +1,50 @@
+import Foundation
+
+struct LibResult<T: Decodable>: Decodable {
+    let status: Int
+    let data: T?
+    let error_message: String?
+
+    func isSuccess() -> Bool {
+        (200...299).contains(status)
+    }
+}
+
+// TODO better way to decode LibResult when it doesn't have data. Maybe don't use Decodable.
+typealias ArbitraryType = String
+
+extension LibResult {
+
+    func toResult() -> Result<T, ApiError> {
+        if isSuccess() {
+            if let data = data {
+                return .success(data)
+            } else {
+                return .failure(.error(message: "Unexpected: Library result success but no data: \(self)"))
```

Comment:
@i-schuetz  Empty response is a valid state. Lets just return .success(()) ?

Reply:
There's `toVoidResult()` for that.